### PR TITLE
ci: run github actions on buildjet

### PR DESCRIPTION
**Pull request by q9f**
Changes introduced:
- Removes CircleCI completely
- Migrates missing jobs to Github Actions
- Introduces powerful runners hosted by BuildJet
- Ensures tests run on all available core (if possible)
- Deduplicates jobs and tasks in Github Actions
- Introduces Rust Version and OS Platform Matrix for builds
- Restores Code Coverage Reports (Tarpaulin/Codecov)
- ~Fixes the Documentation job~ #1383 

Rationale:
- Get this done once and for all
- Ultimate goal: full required pipeline jobs finish < 20 minutes

Further Information:

```yaml
      matrix:
        os: [ubuntu-latest, macos-latest]
        rv: [1.57.0, stable]
```

TODO: @noot please remove CircleCI integration from this repository 🙏🏼  ...
- https://app.circleci.com/settings/project/github/ChainSafe/forest

... and adjust the names of the required tests according to this pull-request. 🙏🏼 🙏🏼 🙏🏼 
- `Rust / Test`
- `Rust / Lint`
- `Rust / Audit`
- `Rust / Build (ubuntu-latest, 1.57.0)`
- `Rust / Build (macos-latest, 1.57.0)`

### CODE REVIEW

**Comment by noot** [comment]
@q9f the tests looks pretty slow (2 hours!) is there a way to speed them up? the current github actions job takes around 45mins, and the circleci job is even faster (I think 20-30mins)

File: Cargo.toml [hunk 1]
```diff
@@ -69,3 +69,6 @@ commcid = { path = "./utils/commcid" }
 forest_vm = { path = "./vm" }
 forest_runtime = { path = "./vm/runtime" }
 forest_message = { path = "./vm/message" }
+
+[profile.dev]
+debug = 0
```
Comment (lemmih) [inline Cargo.toml:74]: Does this improve build times?

Comment (q9f) [reply Cargo.toml:74]: Yes, unless you need debug symbols.

I found this article helpful: https://matklad.github.io/2021/09/04/fast-rust-builds.html

**Comment by q9f** [comment]
This is good to go now.

@noot please adjust the names of the required tests according to this pull-request.
- `Rust / Test`
- `Rust / Lint`
- `Rust / Audit`
- `Rust / Build (ubuntu-latest, 1.57.0)`
- `Rust / Build (macos-latest, 1.57.0)`

**Review by noot — APPROVED** [review]

**Pull request merged**